docs/extension.rst

```diff
@@ -3,30 +3,60 @@ Visual Studio Code Extension
 ============================
 
 Solang has
-`language server <https://en.wikipedia.org/wiki/Language_Server_Protocol>`_ built
-into the executable, which can be used by the Visual Studio Code extension. This
+a `language server <https://en.wikipedia.org/wiki/Language_Server_Protocol>`_ built
+into the executable, which can be used by the Visual Studio Code extension, or by
+any editor that can use a lsp language server. The Visual Studio Code
 extension provides the following:
 
-1. Syntax highlighting
-2. Compiler warnings and errors are displayed in the problems tab and marked
-   with squiqqly lines, this is also known as `diagnostics`.
-3. Hovering over variables, types, functions etc and more will give information,
-   For example this will give the struct fields when hovering over a variable
+1. Syntax highlighting.
+2. Compiler warnings and errors displayed in the problems tab and marked
+   with squiggly lines.
+3. Additional information when hovering over variables, types, functions, etc.
+   For example, this will give the struct fields when hovering over a variable
    which is a reference to a struct.
 
 .. image:: extension-screenshot.png
 
 Both the Visual Studio Code extension code and the language server were developed under a
 `Hyperledger Mentorship programme <https://wiki.hyperledger.org/display/INTERN/Create+a+new+Solidity+Language+Server+%28SLS%29+using+Solang+Compiler>`_.
 
+Solidity Language flavour
+-------------------------
+
+The Solidity language flavour depends on what target you are compiling for. For
+example:
+
+* Ethereum style address literals like ``0xE0f5206BBD039e7b0592d8918820024e2a7437b9`` are
+  not supported on Substrate or Solana, but are supported for EVM.
+* On Substrate and Solana, base58 style encoded address literals like
+  ``address"5GBWmgdFAMqm8ZgAHGobqDqX6tjLxJhv53ygjNtaaAn3sjeZ"`` are supported, but
+  not with EVM.
+* On Solana, there is special builtin import file called ``'solana'`` available.
+* Different blockchains offer different builtins. See the :ref:`builtins documentation <builtins>`.
+* See :ref:`language-status` for compatiblity with Ethereum Solidity (solc)
+* There are many more differences, which are noted throughout the documentation.
+
+You can choose the between the following targets:
+
+solana
+   Solidity for Solana
+
+substrate
+   Solidity for Substrate (Polkadot)
+
+evm
+   Solidity for any EVM based chain like Ethereum
+
+Note that the language server has support for evm, but Hyperledger Solang does
+not support compiling for evm (yet).
```

Comment:
I believe we need to choose whether we want to write EVM in upper case or lower case. You used both ways here. I prefer the upper case. Please, choose only one way to use throughout the documentation.

Reply:
Please note that the there is one lower case evm left in the documentation, because the command line option and the option in the extension settings is lower case evm.